src/test/java/teammates/test/cases/logic/CoursesLogicTest.java

```diff
@@ -438,12 +438,22 @@ private void testGetCoursesForStudentAccount() throws Exception {
         assertEquals(2, courseList.size());
 
         CourseAttributes course1 = dataBundle.courses.get("typicalCourse1");
-        assertEquals(course1.getId(), courseList.get(0).getId());
-        assertEquals(course1.getName(), courseList.get(0).getName());
 
         CourseAttributes course2 = dataBundle.courses.get("typicalCourse2");
-        assertEquals(course2.getId(), courseList.get(1).getId());
-        assertEquals(course2.getName(), courseList.get(1).getName());
+
+        boolean isCourse1Present = false;
+        boolean isCourse2Present = false;
+
+        for (CourseAttributes course : courseList) {
+            if (course.getId().equals(course1.getId()) && course.getName().equals(course1.getName())) {
+                isCourse1Present = true;
+            } else if (course.getId().equals(course2.getId()) && course.getName().equals(course2.getName())) {
+                isCourse2Present = true;
+            }
+        }
+
+        assertTrue(isCourse1Present);
+        assertTrue(isCourse2Present);
```

Comment:
More like 2. We have one such method, but it requires `equals` to be defined which is not the case for any of the `*Attributes` class.

> The current one is just a stop gap solution that does not scale well when the number of objects involved are higher.

I will not say this is a problem. Unscalable as it may, we do not need it to be scalable at this point of time.

> Another approach to consider is to avoid the problem altogether by getting rid of the non-determinism. e.g. modify the API to return the values in a specified sort order.

I won't object to this solution.